lfs/batcher.go

```diff
@@ -6,36 +6,43 @@ import "sync/atomic"
 // be added to the batcher from multiple goroutines and pulled off in groups
 // when one of the following conditions occurs:
 //   * The batch size is reached
+//   * Flush() is called, forcing the batch to be returned immediately, as-is
 //   * Exit() is called
-// When an Exit() occurs, the group may be smaller than the batch size.
+// When an Exit() or Flush() occurs, the group may be smaller than the batch
+// size.
 type Batcher struct {
 	exited     uint32
 	batchSize  int
 	input      chan interface{}
 	batchReady chan []interface{}
+	flush      chan interface{}
 }
 
 // NewBatcher creates a Batcher with the batchSize.
 func NewBatcher(batchSize int) *Batcher {
 	b := &Batcher{
 		batchSize:  batchSize,
-		input:      make(chan interface{}, batchSize),
+		input:      make(chan interface{}),
```

Comment:
Here's what he said in internal chat:

> yeah it looks like it's all done sequentially. if it's cleaner to not buffer, i say do it, 400ns is noise when we're talking network transfers

I agree with the assessment here. Removing buffering lets us keep this to a simple select loop without a noticeable performance impact.